dbt_platform_helper/utils/files.py

```diff
@@ -102,3 +106,69 @@ def combine_env_data(data):
     enriched_config["environments"] = defaulted_envs
 
     return enriched_config
+
+
+def read_supported_versions_from_cache(resource_name):
+
+    platform_helper_config = read_file_as_yaml(PLATFORM_HELPER_CACHE_FILE)
+
+    return platform_helper_config.get(resource_name).get("versions")
+
+
+def write_to_cache(resource_name, supported_versions):
+
+    platform_helper_config = {}
+
+    if os.path.exists(PLATFORM_HELPER_CACHE_FILE):
+        platform_helper_config = read_file_as_yaml(PLATFORM_HELPER_CACHE_FILE)
+
+    cache_dict = {
+        resource_name: {
+            "versions": supported_versions,
+            "date-retrieved": datetime.now().strftime("%d-%m-%y %H:%M:%S"),
+        }
+    }
+
+    platform_helper_config.update(cache_dict)
+
+    with open(PLATFORM_HELPER_CACHE_FILE, "w") as file:
+        file.write("# [!] This file is autogenerated via the platform-helper. Do not edit.\n")
+        yaml.dump(platform_helper_config, file)
+
+
+def cache_refresh_required(resource_name) -> bool:
+    """
+    Checks if the platform-helper should reach out to AWS to 'refresh' its
+    cached values.
+
+    An API call is needed if any of the following conditions are met:
+        1. No cache file (.platform-helper-config.yml) exists.
+        2. The resource name (e.g. redis, opensearch) does not exist within the cache file.
+        3. The date-retrieved value of the cached data is > than a time interval. In this case 1 day.
+    """
+
+    if not os.path.exists(PLATFORM_HELPER_CACHE_FILE):
+        return True
+
+    platform_helper_config = read_file_as_yaml(PLATFORM_HELPER_CACHE_FILE)
+
+    if platform_helper_config.get(resource_name):
+        return check_if_cached_datetime_is_greater_than_interval(
+            platform_helper_config[resource_name].get("date-retrieved"), 1
+        )
+
+    return True
+
+
+def check_if_cached_datetime_is_greater_than_interval(date_retrieved, interval_in_days):
+
+    current_datetime = datetime.now()
+    cached_datetime = datetime.strptime(date_retrieved, "%d-%m-%y %H:%M:%S")
+    delta = current_datetime - cached_datetime
+
+    return False if delta.days < interval_in_days else True
```

Comment:
Could this be...

```suggestion
    return delta.days >= interval_in_days
```

That should give the same behaviour, but the method name would imply...

```suggestion
    return delta.days > interval_in_days
```